### PR TITLE
Remove deprecated JetCorrector and related files

### PR DESCRIPTION
**Pull request by wddgit**
#### PR description:

Remove the deprecated ::JetCorrector class from CMSSW and other related files with it. The migration from ::JetCorrector to reco::JetCorrector started in 2014 and it is time to finally complete it. For the most part this PR removes unused dead code and simplifies things. It makes it impossible for someone to use the deprecated code and obvious failures will occur if they try. It reduces the amount of code we need to maintain and support.

Almost this entire PR consists of deletions. There is one exception to that. The exception is in RecoHI/HiJetAlgos/python/hiCaloJetsForTrk_cff.py. I migrated that file from using the old JetCorrector type to using the new reco::JetCorrector. Somehow that one was missed earlier.

There has been a long effort to migrate from the old JetCorrectors to the new JetCorrectors that has been in progress since 2014. Volker Adler did most of this migration long ago. I've recently migrated all the remaining C++ code. I've also migrated all the Python configuration code that is used in existing tests that I can find. And I will help if merging this PR reveals more cases of code in CMSSW that needs migration. There are 4 PRs that are submitted but not merged yet. These need to be merged before this PR or there will be compilation errors. These are #39875 #39864 #39847 and #39804.

I also deleted the obsolete central cff configuration files commonly imported to use the old JetCorrector class.

There are dozens of other configuration files that still reference the old cff files deleted by this PR. Those will be broken by this PR, but they are all old and look to be old obsolete code that is already broken. It is not worth the effort to deal with them at this point. If someone tries to run them they will fail with an obvious exception and then they can migrated to use the new JetCorrector class. There is a good chance none of them will ever be used again.

#### PR validation:

Relies on existing tests. Errors caused by this should result in obvious exceptions.

The one exception is the change in hiCaloJetsForTrk_cff.py. That is tested by runTheMatrix.py test 140.53. That is how I noticed it still needed to be migrated.

If other failures show up, I will be available to help migrate the affected code.

### CODE REVIEW

**Comment by wddgit** [comment]
please test with #39875,#39864,#39847,#39804 

FYI @Dr15Jones @makortel

File: JetMETCorrections/Objects/interface/JetCorrector.h [hunk 1]
```diff
@@ -1,64 +1 @@
...
-
-  /// retrieve corrector from the event setup. troughs exception if something is missing
-  static const JetCorrector* getJetCorrector(const std::string& fName, const edm::EventSetup& fSetup);
-};
-
-// inline method
-inline bool JetCorrector::vectorialCorrection() const { return false; }
-
-#endif
+#error, obsolete header, use reco::JetCorrector in JetMETCorrections/JetCorrector/interface/JetCorrector.h instead
```
Comment (makortel) [inline JetMETCorrections/Objects/interface/JetCorrector.h:1]: Should this be 
```suggestion
#error "obsolete header, use reco::JetCorrector in JetMETCorrections/JetCorrector/interface/JetCorrector.h instead"
```
?

Comment (wddgit) [reply JetMETCorrections/Objects/interface/JetCorrector.h:1]: I will fix that as soon as the tests finish. Thanks. No idea why I put a comma there...

**Comment by makortel** [comment]
> -1
> 
> **Failed Tests**: HeaderConsistency

Note that the header consistency check failure is caused by the `#error`, i.e. it is expected.

**Comment by wddgit** [comment]
please test with #39875,#39864,#39847,#39804

with fix to "#error" line

**Comment by wddgit** [comment]
@smuzaffar Is there a way to make "scram b check-headers" ignore the file where I added "#error"? Would it be better if I just deleted the file instead? The only value it adds is a small message, but usage of the header is already completely removed from CMSSW, so it is not that important. I would hate to have this error appearing every time we have a PR including this package.

**Comment by wddgit** [comment]
In my local working area, I ran the the full complete set of runTheMatrix.py tests, about 1000 of them. All passed except there were 2 unrelated failures ("TritonService: Starting the fallback server failed with exit code 65280" which has nothing to do with JetCorrectors).

I'm leaning towards just deleting the header file instead of putting "#error" in there... And still waiting on the 4 PRs this depends on. Please don't merge this PR until those are also merged.

**Comment by wddgit** [comment]
please test

I have not seen this unit test failure before, TestDQMOfflineConfiguration_150. It not obvious from the log what the problem is. Maybe a glitch. Try again.

**Comment by wddgit** [comment]
abort test

**Comment by wddgit** [comment]
please test with #39875, #39864, #39847, #39804

I have not seen this unit test failure before, TestDQMOfflineConfiguration_150. It not obvious from the log what the problem is. Maybe a glitch. Try again.

**Comment by wddgit** [comment]
please test with #39875,#39864,#39847,#39804

Try again, unit tests passed last time, but now relVal tests which passed before timed out, does not look related to my changes although it is hard to understand why they failed

**Comment by emanueleusai** [comment]
+1

- I agree with the strategy to let the obsolete code as it is. In DQM we just contacted all PAGs and POGs to check which set of monitoring elements are actually being used. We should be able to clean up a good number of the obsolete modules in DQM/Physics once we get an answer from all the groups

**Comment by swertz** [comment]
+xpog

**Comment by clacaputo** [comment]
+reconstruction

- PR removes deprecated JetCorrector 
- #39875,#39864,#39847,#39804 are related to this one. These need to be merged before or there will be compilation errors
- no reco changes observed

**Comment by perrotta** [comment]
Note for the release managers: as of now this still need https://github.com/cms-sw/cmssw/pull/39847 and https://github.com/cms-sw/cmssw/pull/39804 to be merged. This PR will follow

**Comment by wddgit** [comment]
I closed #39804 and replaced it with #40139. So now don't merge this until both #40139 and #39847 are merged. Should I just put this on hold to reduce confusion?

**Comment by perrotta** [comment]
> Should I just put this on hold to reduce confusion?

Yes, why not David?
I'd let you put on hold, instead of doing it myself, so that you can unhold yourself as soon as all other pieces are ready

**Comment by wddgit** [comment]
hold

**Comment by makortel** [comment]
hold

(seems that holding needs more power)

**Comment by wddgit** [comment]
please test

@makortel I think we can remove the hold. I'm rerunning the tests to see if they will pass with no other PR's included. I think it should now that all the other PRs were merged.

**Comment by makortel** [comment]
The comparison differences are in 11634.7, so this looks like another symptom of https://github.com/cms-sw/cmssw/issues/39803#issuecomment-1337009470

**Comment by rappoccio** [comment]
+1

Comparison differences are spurious.

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
@wddgit after the merge of this PR there are unit test errors in Configuration/DataProcessing, apparently related to the relics JetCorrectionServicesAllAlgos_cff.py which is still imported in
- HeavyIonsAnalysis/Configuration/python/HI_DiJetSkim_cff.py
- HeavyIonsAnalysis/Configuration/python/analysisFilters_cff.py

In reality I see many more places in CMSSW where that config is imported, see e.g. https://cmssdt.cern.ch/lxr/search?%21v=CMSSW_13_0_X_2022-12-12-1100&_filestring=&_string=rrectionServicesAllAlgos_cff

Could you please have a look?

**Comment by wddgit** [comment]
I'll take a look this morning.

**Comment by wddgit** [comment]
The PR fixing the unit test is #40301, recently submitted.

I also submitted a new Issue with the list of files importing the deleted cff files. Issue #40304. Note that none of these are used in tests. As far as I know they are not used at all.

**Comment by wddgit** [comment]
Starting with the IB after this was merged, CMSSW_13_0_X_2022-12-12-1100, the Static Analyzer test in the IB named "SA EventSetupRecord::get called" is now reporting this:

"no-package: {}"

I presume this means it found no problems. Clearing that was the main goal of this PR.

Also after the #40301 was merged last night, the unit tests all passed.

**Comment by perrotta** [comment]
Good. Thank you @wddgit , also for the github issue

**Pull request merged**